### PR TITLE
Enable/disable scroll bars on individual tabs.

### PR DESCRIPTION
**Pull request by 12089897411**
Allow individual tabs to control scrolling. Now,the scrolling of each tab is controlled by themselves !

### CODE REVIEW

File: src/widgets/tab_viewer.rs [hunk 1]
```diff
@@ -76,4 +76,14 @@ pub trait TabViewer {
     fn clear_background(&self, _tab: &Self::Tab) -> bool {
         true
     }
+    /// The scrolling of each tab is controlled by themselves
+
+    fn tab_scroll(&self, _tab: &Self::Tab) -> bool {
```
Comment (Adanos020) [inline src/widgets/tab_viewer.rs:81]: The same effect can be achieved by returning `[false, false]` from `scroll_both`, making this method a duplicate. I'd remove `tab_scroll` completely.

File: src/widgets/dock_area/mod.rs [hunk 2]
```diff
@@ -1063,8 +1063,8 @@ impl<'tree, Tab> DockArea<'tree, Tab> {
                 tabs_style.tab_body.stroke,
             );
 
-            if self.scroll_area_in_tabs {
-                ScrollArea::both().show(ui, |ui| {
+            if self.scroll_area_in_tabs && tab_viewer.tab_scroll(tab) {
```
Comment (Adanos020) [inline src/widgets/dock_area/mod.rs:1066]: Once you delete `tab_scroll`, you can delete this `if`, the whole `else` branch, and also the `scoll_area_in_tabs` altogether. The presence of scroll bars will then only be controlled by `TabViewer::scroll_both`.

File: src/widgets/tab_viewer.rs [hunk 3]
```diff
@@ -76,4 +76,14 @@ pub trait TabViewer {
...
+        false
+    }
+    /// Is the tab's scrolling mode both, horizontal, or
+
+    fn scroll_both(&self, _tab: &Self::Tab) -> [bool; 2] {
```
Comment (Adanos020) [inline src/widgets/tab_viewer.rs:86]: I'd rename this method to `scroll_bars`.

File: src/widgets/tab_viewer.rs [hunk 4]
```diff
@@ -76,4 +76,14 @@ pub trait TabViewer {
     fn clear_background(&self, _tab: &Self::Tab) -> bool {
         true
     }
+    /// The scrolling of each tab is controlled by themselves
+
+    fn tab_scroll(&self, _tab: &Self::Tab) -> bool {
+        false
+    }
+    /// Is the tab's scrolling mode both, horizontal, or
```
Comment (Adanos020) [inline src/widgets/tab_viewer.rs:84]: Move this one line down. I'd also rewrite this comment to "Determines if the horizontal and vertical scroll bars are shown for \`tab\`."

**Review by Adanos020 — CHANGES_REQUESTED** [review]
This is a great feature to have but I'd like this implemented a bit differently.

Since now we have a way to control the presence of scroll bars on individual tabs (the `scroll_both` method), there's no reason to have `DockArea::scroll_area_in_tabs` anymore. The `tab_scroll` is also redundant, because the same effect can be achieved by just returning `[false, false]` in `scroll_both`.

I've included more details in the comments.

File: src/widgets/tab_viewer.rs (hunk 4 above)
Comment (12089897411) [reply src/widgets/tab_viewer.rs]: Of course, I removed the qscrollarea in_tabs(), scroll_area_in_tabs parameter and tab scroll(), and I renamed scroll_both() to scroll_bars()！

File: src/widgets/tab_viewer.rs [hunk 5]
```diff
@@ -76,4 +76,10 @@ pub trait TabViewer {
     fn clear_background(&self, _tab: &Self::Tab) -> bool {
         true
     }
+
+    /// if the horizontal and vertical scroll bars are shown for `tab`.
```
Comment (Adanos020) [inline src/widgets/tab_viewer.rs:80]: Capitalize the "if"

**Review by Adanos020 — CHANGES_REQUESTED** [review]
Looks a lot better! Can you also include this change in CHANGELOG.md?

**Review by Adanos020 — APPROVED** [review]
I applied the changes I suggested earlier. For the future, please don't make changes directly to release-0.7, instead create a new branch from it and push your changes there. The reason I'm asking this is because it makes it very difficult for others to submit changes to your PR.

**Pull request merged**